phpunit/navigation-block-fallback-test.php

```diff
@@ -0,0 +1,320 @@
+<?php
+/**
+ * Tests Fallback Behavior for Navigation block
+ *
+ * @package WordPress
+ */
+
+class Tests_Block_Navigation_Fallbacks extends WP_UnitTestCase {
+
+	public function set_up() {
+		parent::set_up();
+		// Navigation menu will be created on Theme switch. Therefore in order to test
+		// the behaviour of `gutenberg_block_core_navigation_create_fallback`
+		// the auto-creation of a fallback must be disabled for this initial
+		// theme switch.
+		add_filter( 'block_core_navigation_skip_fallback', '__return_true' );
+		switch_theme( 'emptytheme' );
+		remove_filter( 'block_core_navigation_skip_fallback', '__return_true' );
+	}
+
+	private function get_navigations_in_database() {
+		$navs_in_db = new WP_Query(
+			array(
+				'post_type'      => 'wp_navigation',
+				'post_status'    => 'publish',
+				'posts_per_page' => -1,
+				'orderby'        => 'date',
+				'order'          => 'DESC',
+			)
+		);
+
+		return $navs_in_db->posts ? $navs_in_db->posts : array();
+	}
+
+	private function mock_wp_install() {
+		$user = get_current_user();
+		do_action( 'wp_install', $user );
+	}
+
+	public function test_should_auto_create_navigation_menu_on_wp_install() {
+		$this->mock_wp_install();
+
+		$navs_in_db = $this->get_navigations_in_database();
+		$this->assertCount( 1, $navs_in_db );
+	}
+
+	public function test_should_auto_create_navigation_menu_on_theme_switch() {
+		// Remove any existing disabling filters.
+		remove_filter( 'block_core_navigation_skip_fallback', '__return_true' );
+
+		$navs_in_db = $this->get_navigations_in_database();
+		$this->assertCount( 0, $navs_in_db );
```

Comment:
If more than one assertion is used in a test method, the `$message` parameter should be passed to the each assertion method that supports it.

https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions

For example:
```suggestion
		$this->assertCount( 0, $navs_in_db, 'Navigation Menu should not exist before running the tests.' );
```